apps/admin-gui/src/app/vos/pages/member-detail-page/member-overview/member-overview.component.ts

```diff
@@ -144,6 +160,44 @@ export class MemberOverviewComponent implements OnInit {
     });
   }
 
+  sponsorThisMember(): void {
+    const config = getDefaultDialogConfig();
+    config.width = '400px';
+    config.data = {
+      theme: 'member-theme',
+      member: this.member,
+      sponsors: this.voSponsors.filter(
+        (s) => !this.sponsors.map((sponsor) => sponsor.user.id).includes(s.id)
+      ),
+    };
+
+    const dialogRef = this.dialog.open(SponsorThisMemberDialogComponent, config);
+
+    dialogRef.afterClosed().subscribe((edited) => {
+      if (edited) {
+        this.loading = true;
+        this.membersService.getRichMemberWithAttributes(this.member.id).subscribe((member) => {
+          this.member = member;
+          this.usersManager.getSponsorsForMember(this.member.id, null).subscribe((sponsors) => {
+            this.sponsors = sponsors;
+            this.sponsorsDataSource.data = this.sponsors;
+          });
+          this.loading = false;
+        });
+      }
```

Comment:
This chunk of code is duplicate, please extract this logic and move it to a separate method (`refreshData()` seems like a good place)